lifetimes/utils.py

```diff
@@ -109,11 +109,12 @@ def to_period(d):
 
     transactions = transactions.ix[transactions[datetime_col] <= observation_period_end]
 
-    # reduce all events per customer during the period to a single event:
-    period_transactions = reduce_events_to_period(transactions, *select_columns).reset_index(level=select_columns[1:])
-
     # count all orders by customer.
-    customers = period_transactions[datetime_col].groupby(level=customer_id_col, sort=False).agg(['max', 'min', 'count'])
+    customers = transactions.groupby(customer_id_col).apply(
+        lambda x: pd.Series({'max': x[datetime_col].max(),
+                            'min': x[datetime_col].min(),
+                            'count': len(x[datetime_col].unique())
```

Comment:
it's the same when *select_columns is just [customer_id_col, datetime_col]

but when *select_columns is [customer_id_col, datetime_col, monetary_value_col], the original statement will include monetary_value_col in the group such that 'same date different amount' transactions will not be grouped - which I believe that is not what it intends to be.

As I haven't constructed the period_transactions dataframe here, I just rewrite the groupby statement, and since simply `agg('count')` won't group the transactions in the same date as a single transaction, therefore I replace it with `lambda x: len(x[datetime_col].unique())`.